Marlin/src/inc/SanityCheck.h

```diff
@@ -1445,8 +1445,14 @@ static_assert(Y_MAX_LENGTH >= Y_BED_SIZE, "Movement bounds (Y_MIN_POS, Y_MAX_POS
     #error "AUTO_BED_LEVELING_UBL requires EEPROM_SETTINGS."
   #elif !WITHIN(GRID_MAX_POINTS_X, 3, 15) || !WITHIN(GRID_MAX_POINTS_Y, 3, 15)
     #error "GRID_MAX_POINTS_[XY] must be a whole number between 3 and 15."
-  #elif !defined(RESTORE_LEVELING_AFTER_G28) && !defined(ENABLE_LEVELING_AFTER_G28)
-    #error "AUTO_BED_LEVELING_UBL used to enable RESTORE_LEVELING_AFTER_G28. To keep this behavior enable RESTORE_LEVELING_AFTER_G28. Otherwise define it as 'false'."
+  #elif !defined(LEVELING_AFTER_G28)
+    #error "AUTO_BED_LEVELING_UBL used to set LEVELING_AFTER_G28=ACTIVATE_LVL. To keep this behavior set LEVELING_AFTER_G28=ACTIVATE_LVL. Otherwise define it as 'DEACTIVE_LVL'."
+  #elif LEVELING_AFTER_G28 != DEACTIVATE_LVL && LEVELING_AFTER_G28 != ACTIVATE_LVL && LEVELING_AFTER_G28 != RESTORE_LVL
```

Comment:
Should there be a check if `LEVELING_AFTER_G28` has been enabled without a value or the default error message is enough? `error: operator '!=' has no left operand`

Reply:
The check you are talking about is just 2 lines below thwe one you report here:
```c++
#elif LEVELING_AFTER_G28 != DEACTIVATE_LVL && LEVELING_AFTER_G28 != ACTIVATE_LVL && LEVELING_AFTER_G28 != RESTORE_LVL
    #error "LEVELING_AFTER_G28 must be an option between DEACTIVATE_LVL, ACTIVATE_LVL or RESTORE_LVL."
```

Reply:
I clicked on the right line, but it only shows as the last line, but yes, these checks are what I was talking about:
https://github.com/MarlinFirmware/Marlin/blob/544b867db24481159326fdc280515051c66da6d2/Marlin/src/inc/SanityCheck.h#L1450-L1455

Reply:
Then I don't understand the first post. Value must be a valid value and if undefined there is an old sanity check that I have updated

Reply:
If `LEVELING_AFTER_G28` is defined, but don't have a value (as it is by default), then we will get `error: operator '!=' has no left operand`. My comment was about, if that error message is ok or not, but if the error message isn't good, and if there is no way to check if a variable is defined without any value, then the problem could be mitigated if `LEVELING_AFTER_G28` have a value (e.g. `RESTORE_LVL`) by default.

Reply:
ok now I understood, thanks

Reply:
Sorry about the misunderstanding it was my fault.

Reply:
Oh, really sometimes I have some problem to understand obviously English sentences, it's not my language ... well sometime I do even with mine  :-D